contracts/SportMarkets/MarchMadness/MarchMadness.sol

```diff
@@ -0,0 +1,125 @@
+
+    mapping(uint => uint[61]) public itemToBrackets;
+    mapping(address => uint) public addressToTokenId;
+
+    /* ========== CONSTRUCTOR ========== */
+
+    constructor() ERC721(_name, _symbol) {}
+
+    /* ========== OWC ========== */
+
+    function mint(uint[61] memory _brackets) external whenNotPaused returns (uint newItemId) {
+        require(addressAlreadyMinted[msg.sender] == false, "Address already minted");
+        require(block.timestamp > canNotMintOrUpdateBefore, "Can not mint before settled date");
+        require(block.timestamp < canNotMintOrUpdateAfter, "Can not mint after settled date");
+
+        _tokenIds.increment();
+
+        newItemId = _tokenIds.current();
+
+        _mint(msg.sender, newItemId);
+
+        addressAlreadyMinted[msg.sender] = true;
+
+        itemToBrackets[newItemId] = _brackets;
+        addressToTokenId[msg.sender] = newItemId;
+
+        emit Mint(msg.sender, newItemId, _brackets);
+    }
+
+    function updateBracketsForAlreadyMintedItem(uint _tokenId, uint[61] memory _brackets) external whenNotPaused {
+        require(_exists(_tokenId), "Item does not exists");
+        require(ownerOf(_tokenId) == msg.sender, "Caller is not owner of entered tokenId");
+        itemToBrackets[_tokenId] = _brackets;
+        emit UpdateBracketsForAlreadyMintedItem(msg.sender, _tokenId, _brackets);
+    }
+
+    /* ========== VIEW ========== */
+
+    function getCorrectPositionsByTokenId(uint _tokenId) public view returns (uint) {
+        uint[61] memory _brackets = itemToBrackets[_tokenId];
+        if (_brackets[1] == 0) return 0;
+
+        uint correctPredictions = 0;
+
+        for (uint i = 0; i < _brackets.length; i++) {
+            if (_brackets[i] == results[i]) {
+                correctPredictions++;
+            }
+        }
+
+        return correctPredictions;
+    }
+
+    function getCorrectPositionsByMinterAddress(address _minter) public view returns (uint) {
+        if (addressToTokenId[_minter] == 0) return 0;
+        return getCorrectPositionsByTokenId(addressToTokenId[_minter]);
+    }
+
+    /* ========== INTERNALS ========== */
+
+    function _beforeTokenTransfer(
+        address from,
+        address to,
+        uint256 tokenId
+    ) internal virtual override {
+        require(from == address(0) || to == address(0), "NonTransferrableERC721Token: non transferrable");
+    }
+
+    /* ========== CONTRACT MANAGEMENT ========== */
+
+    function pause() public onlyOwner {
+        _pause();
+    }
+
+    function unpause() public onlyOwner {
+        _unpause();
+    }
+
+    function setResultForGame(uint _game, uint _teamIndex) external onlyOwner {
+        require(_teamIndex < 69, "Not valid team index");
+        require(_game < 61, "Not valid game index");
+        results[_game] = _teamIndex;
+        emit ResultForGameAdded(_game, _teamIndex);
+    }
+
+    function setDateRange(uint _fromDate, uint _toDate) external onlyOwner {
```

Comment:
some `require` statements  like `_fromDate < _toDate`

Reply:
I think it's no needed, because we remove beforeDate